### PR TITLE
feat(automation anywhere): implement connector

### PR DESCRIPTION
**Pull request by Oleksiivanov**
## Description

Implemented code for automation anywhere connector. 

Related : https://github.com/camunda/team-connectors/issues/521

### CODE REVIEW

**Comment by igpetrov** [comment]
It looks to me that it's an HTTP connector. Why would we need to implement it in code then?

**Comment by sbuettner** [comment]
@igpetrov We want to get rid of the custom authentication method in the http client to simplify it. Hence the need for custom code.

**Comment by igpetrov** [comment]
> @igpetrov We want to get rid of the custom authentication method in the http client to simplify it. Hence the need for custom code.

@sbuettner @Oleksiivanov oh, I see. that's sad

File: connectors/automation-anywhere/src/main/java/io/camunda/connector/automationanywhere/AutomationAnywhereConnector.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      "url",
+      "method",
+      "authentication",
+      "headers",
+      "queryParameters",
+      "connectionTimeoutInSeconds",
+      "body"
+    },
+    type = "io.camunda:connector-automationanywhere:1")
+public class AutomationAnywhereConnector implements OutboundConnectorFunction {
```
Comment (sbuettner) [inline connectors/automation-anywhere/src/main/java/io/camunda/connector/automationanywhere/AutomationAnywhereConnector.java:35]: Can you please use the element template generator?

File: connectors/automation-anywhere/src/main/java/io/camunda/connector/automationanywhere/model/AutomationAnywhereRequest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.automationanywhere.model;
+
+import io.camunda.connector.automationanywhere.model.auth.AuthenticationStrategy;
+import io.camunda.connector.http.base.model.HttpCommonRequest;
+import jakarta.validation.Valid;
+
+public class AutomationAnywhereRequest extends HttpCommonRequest {
```
Comment (sbuettner) [inline connectors/automation-anywhere/src/main/java/io/camunda/connector/automationanywhere/model/AutomationAnywhereRequest.java:13]: Why do you need to inherit here? Cant you just "build" the required http common request in code?

**Review by sbuettner — CHANGES_REQUESTED** [review]
Please use element template generator and dont inherit model from base.

File: connectors/automation-anywhere/src/main/java/io/camunda/connector/automationanywhere/AutomationAnywhereConnector.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  private static final String AUTHENTICATION_HEADER = "X-Authorization";
```
Comment (igpetrov) [inline connectors/automation-anywhere/src/main/java/io/camunda/connector/automationanywhere/AutomationAnywhereConnector.java:36]: Consider using `protected` in order to re-use constant in tests.

```suggestion
  protected static final String AUTHENTICATION_HEADER = "X-Authorization";
```

File: connectors/automation-anywhere/src/main/java/io/camunda/connector/automationanywhere/model/auth/PasswordBasedAuthenticationStrategy.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      final ObjectMapper objectMapper)
+      throws IOException, InstantiationException, IllegalAccessException {
+
+    final var authRequest = createAuthRequest(url, createRequestBody(), connectionTimeoutInSeconds);
+    final var result = httpService.executeConnectorRequest(authRequest);
+    return fetchToken(result, objectMapper);
+  }
+
+  private Map<String, Object> createRequestBody() {
+    return Map.of("username", username, "password", password, "multipleLogin", multipleLogin);
```
Comment (igpetrov) [inline connectors/automation-anywhere/src/main/java/io/camunda/connector/automationanywhere/model/auth/PasswordBasedAuthenticationStrategy.java:36]: Here as well

**Review by igpetrov — CHANGES_REQUESTED** [review]
Minor changes requested.

File: connectors/automation-anywhere/src/main/java/io/camunda/connector/automationanywhere/AutomationAnywhereConnector.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    inputDataClass = AutomationAnywhereRequest.class,
+    version = 1,
+    propertyGroups = {
+      @ElementTemplate.PropertyGroup(id = "operation", label = "Operation"),
+      @ElementTemplate.PropertyGroup(id = "configuration", label = "Configuration"),
+      @ElementTemplate.PropertyGroup(id = "authentication", label = "Authentication"),
+      @ElementTemplate.PropertyGroup(id = "input", label = "Input"),
+      @ElementTemplate.PropertyGroup(id = "timeout", label = "Timeout")
+    },
+    documentationRef = "https://docs.camunda.io/docs/components/connectors/protocol/rest/",
```
Comment (igpetrov) [inline connectors/automation-anywhere/src/main/java/io/camunda/connector/automationanywhere/AutomationAnywhereConnector.java:41]: Doesn't it have it's own doc?

**Review by igpetrov — APPROVED** [review]
LGTM

**Review by sbuettner — APPROVED** [review]
We are missing proper e2e tests. Please add them in another PRs.

**Pull request merged**